src/components/NewMovie/NewMovie.js

```diff
@@ -1,13 +1,238 @@
+import './NewMovie.scss';
+import PropTypes from 'prop-types';
+
+// eslint-disable-next-line
+const regexp = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/;
 
 export class NewMovie extends Component {
-  state = {};
+  state = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+
+    errorTitle: '',
+    errorImgUrl: '',
+    errorImdbUrl: '',
+    errorImdbId: '',
+  };
+
+  handleSetTitle = (e) => {
+    this.setState({
+      title: e.target.value,
+      errorTitle: '',
+    });
+  }
+
+  handleSetDescription = (e) => {
+    this.setState({
+      description: e.target.value,
+    });
+  }
+
+  handleSetImgUrl = (e) => {
+    this.setState({
+      imgUrl: e.target.value,
+      errorImgUrl: '',
+    });
+  }
+
+  handleSetImdbUrl = (e) => {
+    this.setState({
+      imdbUrl: e.target.value,
+      errorImdbUrl: '',
+    });
+  }
+
+  handleSetImdbId = (e) => {
+    this.setState({
+      imdbId: e.target.value,
+      errorImdbId: '',
+    });
+  }
+
+  resetForm = () => {
+    this.setState({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+
+      errorTitle: '',
+      errorImgUrl: '',
+      errorImdbUrl: '',
+      errorImdbId: '',
+    });
+  }
+
+  hasErrorTitle = () => {
+    this.setState((prev) => {
+      const nextState = {};
+
+      if (prev.title.trim() === '') {
+        nextState.errorTitle = true;
+      }
+
+      return nextState;
+    });
+  }
+
+  hasErrorImgUrl = () => {
+    this.setState((prev) => {
+      const nextState = {};
+
+      if (!regexp.test(prev.imgUrl)) {
+        nextState.errorImgUrl = true;
+      }
+
+      return nextState;
+    });
+  }
+
+  hasErrorimdbUrl = () => {
+    this.setState((prev) => {
+      const nextState = {};
+
+      if (!regexp.test(prev.imdbUrl)) {
+        nextState.errorImdbUrl = true;
+      }
+
+      return nextState;
+    });
+  }
+
+  hasErrorImdbId = () => {
+    this.setState((prev) => {
+      const nextState = {};
+
+      if (prev.imdbId.trim() === '') {
+        nextState.errorImdbId = true;
+      }
+
+      return nextState;
+    });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (this.state.title.trim() !== ''
+      && regexp.test(this.state.imgUrl)
+      && regexp.test(this.state.imdbUrl)
+      && this.state.imdbId.trim() !== '') {
+      this.props.addMovie(
+        this.state.title,
+        this.state.description,
+        this.state.imgUrl,
+        this.state.imdbUrl,
+        this.state.imdbId,
+      );
+
+      this.resetForm();
+    }
+  }
 
   render() {
+    const {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+      errorTitle,
+      errorImgUrl,
+      errorImdbUrl,
+      errorImdbId,
+    } = this.state;
+
+    const disableButton = errorTitle === ''
```

Comment:
```suggestion
    const buttonState = errorTitle === ''
```
names with verb phrases are for functions